components/certificate-mgt/org.wso2.carbon.identity.certificate.management.core/src/main/java/org/wso2/carbon/identity/certificate/management/core/dao/impl/ApplicationCertificateManagementDAOImpl.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.certificate.management.core.dao.impl;
+
+import org.wso2.carbon.database.utils.jdbc.NamedJdbcTemplate;
+import org.wso2.carbon.database.utils.jdbc.exceptions.DataAccessException;
+import org.wso2.carbon.database.utils.jdbc.exceptions.TransactionException;
+import org.wso2.carbon.identity.certificate.management.core.constant.ApplicationCertificateMgtSQLQueries;
+import org.wso2.carbon.identity.certificate.management.core.constant.CertificateMgtErrors;
+import org.wso2.carbon.identity.certificate.management.core.constant.CertificateMgtSQLConstants;
+import org.wso2.carbon.identity.certificate.management.core.dao.ApplicationCertificateManagementDAO;
+import org.wso2.carbon.identity.certificate.management.core.exception.CertificateMgtException;
+import org.wso2.carbon.identity.certificate.management.core.model.Certificate;
+import org.wso2.carbon.identity.certificate.management.core.util.CertificateMgtUtil;
+import org.wso2.carbon.identity.core.util.IdentityDatabaseUtil;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This class is the implementation of the Application Certificate Management DAO.
+ * @deprecated Use {@link CertificateManagementDAOImpl} instead.
+ */
+public class ApplicationCertificateManagementDAOImpl implements ApplicationCertificateManagementDAO {
+
+    @Override
+    @Deprecated
+    public int addCertificate(String certificateId, Certificate certificate, int tenantId)
+            throws CertificateMgtException {
+
+        NamedJdbcTemplate jdbcTemplate = new NamedJdbcTemplate(IdentityDatabaseUtil.getDataSource());
+        try {
+            InputStream certByteStream = CertificateMgtUtil.getCertificateByteStream(certificate.getCertificate());
+            int certificateLength = certByteStream.available();
+            return jdbcTemplate.withTransaction(template ->
+                    template.executeInsert(CertificateMgtSQLConstants.Query.ADD_CERTIFICATE,
+                            preparedStatement -> {
+                                int index = 1;
+                                preparedStatement.setString(index++, certificateId);
```

Comment:
can't we use named prepared statements ?

Reply:
Since we are using simpler queries preparedStatement is the most suitable than the NamedPreparedStatement